app/dashboard/courses/MyCoursesList.cy.tsx

```diff
@@ -0,0 +1,67 @@
+import { MyCoursesList } from "./MyCoursesList";
+import * as nextAuth from "next-auth/react";
+import * as reactQuery from "@tanstack/react-query";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+
+describe("<MyCoursesList />", () => {
+  const queryClient = new QueryClient();
+  it("should show loading state when data is still being fetched", () => {
+    cy.stub(nextAuth, "useSession").returns({
+      data: { userId: "user123" },
+      status: "authenticated",
+    });
+
+    cy.stub(reactQuery, "useQuery").returns({
+      data: null,
+      isLoading: true,
+    });
+    cy.mount(
+      <QueryClientProvider client={queryClient}>
+        <MyCoursesList />
+      </QueryClientProvider>
+    );
+    cy.contains("Loading...").should("exist");
+  });
+
+  it("should not render courses if session is invalid", () => {
+    cy.stub(nextAuth, "useSession").returns({
+      data: null,
+      status: "unauthenticated",
+    });
+    cy.stub(reactQuery, "useQuery").returns({
+      data: [],
+      isLoading: false,
+    });
+    cy.mount(
+      <QueryClientProvider client={queryClient}>
+        <MyCoursesList />
+      </QueryClientProvider>
+    );
```

Comment:
you can extract such logic to separate function and reuse between these tests. It is not critical, but test will be more clear with no duplicates. Use .beforeEach and .afterEach and more abstract functions (for example renderCoursesList(queryProvider) instead of few lines of just mounting) for such purposes